process/block/preprocess/basePreProcess.go

```diff
@@ -415,3 +415,21 @@ func (bpp *basePreProcess) getTxMaxTotalCost(txHandler data.TransactionHandler)
 
 	return cost
 }
+
+func (bpp *basePreProcess) getTotalGasConsumed() uint64 {
+	totalGasConsumed := uint64(0)
+	if bpp.gasHandler.TotalGasConsumed() > bpp.gasHandler.TotalGasRefunded() {
+		totalGasConsumed = bpp.gasHandler.TotalGasConsumed() - bpp.gasHandler.TotalGasRefunded()
+	}
+
+	return totalGasConsumed
+}
+
+func (bpp *basePreProcess) updateGasConsumedWithGasRefunded(
+	txHash []byte,
+	gasConsumedByMiniBlockInReceiverShard *uint64,
```

Comment:
let's not use these pointer input arguments as outputs. Maybe create a struct that will wrap these 3 params and pass it as pointer and change the values inside? The function states that an update will occur but using uint pointers is a little bit ugly.

Reply:
This is done in a big refactor which has been occurred over this component in scheduled txs feature.